drivers/video/video_buffer.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2019, Linaro Limited
+ * Copyright (c) 2024-2025, tinyVision.ai Inc.
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/video.h>
+#include <zephyr/rtio/rtio.h>
+
+#include "video_buffer.h"
+#include "video_device.h"
+
+LOG_MODULE_REGISTER(video_buffer, CONFIG_VIDEO_LOG_LEVEL);
+
+#if defined(CONFIG_VIDEO_BUFFER_USE_SHARED_MULTI_HEAP)
+#include <zephyr/multi_heap/shared_multi_heap.h>
+
+#define VIDEO_COMMON_HEAP_ALLOC(align, size, timeout)                                              \
+	shared_multi_heap_aligned_alloc(CONFIG_VIDEO_BUFFER_SMH_ATTRIBUTE, align, size)
+#define VIDEO_COMMON_FREE(block) shared_multi_heap_free(block)
+#else
+K_HEAP_DEFINE(video_buffer_pool, CONFIG_VIDEO_BUFFER_POOL_SZ_MAX *CONFIG_VIDEO_BUFFER_POOL_NUM_MAX);
+#define VIDEO_COMMON_HEAP_ALLOC(align, size, timeout)                                              \
+	k_heap_aligned_alloc(&video_buffer_pool, align, size, timeout);
+#define VIDEO_COMMON_FREE(block) k_heap_free(&video_buffer_pool, block)
+#endif
```

Comment:
Given the new allocation implementation does not use most of the parameters, this block could be simplified:

```suggestion
#if defined(CONFIG_VIDEO_BUFFER_USE_SHARED_MULTI_HEAP)

/* Wrappers around Shared MultiHeap (SMH) API */

#include <zephyr/multi_heap/shared_multi_heap.h>

#define VIDEO_COMMON_ALLOC(size)                                                                   \
	shared_multi_heap_aligned_alloc(CONFIG_VIDEO_BUFFER_SMH_ATTRIBUTE,                         \
					CONFIG_VIDEO_BUFFER_POOL_ALIGN, size)

#define VIDEO_COMMON_FREE(ptr) shared_multi_heap_free(ptr)

#else

/* Wrappers around k_heap() APIs */

K_HEAP_DEFINE(video_buffer_pool,                                                                   \
	      CONFIG_VIDEO_BUFFER_POOL_SZ_MAX * CONFIG_VIDEO_BUFFER_POOL_NUM_MAX);

#define VIDEO_COMMON_ALLOC(size)                                                                   \
	k_heap_aligned_alloc(&video_buffer_pool, CONFIG_VIDEO_BUFFER_POOL_ALIGN, size, K_NO_WAIT)

#define VIDEO_COMMON_FREE(block) k_heap_free(&video_buffer_pool, block)

#endif
```